src/test/java/com/wikia/webdriver/common/core/url/UrlBuilder.java

```diff
@@ -1,145 +1,94 @@
 
-  public String getLanguageForWiki() {
-    return Configuration.getWikiLanguage() == null ? Configuration.DEFAULT_LANGUAGE : Configuration.getWikiLanguage();
+  public static UrlBuilder createUrlBuilderForWikiAndLang(String wiki, String language) {
+    return new UrlBuilder(wiki, getEnv(), getForceHttps(), getForceLanguageInPath(), language);
   }
 
   public String normalizePageName(String pageName) {
     return pageName.replace(" ", "_");
   }
 
   public String getUrlForPageWithWWW(String pageName) {
-    return getUrlForWiki(Configuration.getWikiName(), true) + pageName;
+    return getUrlForWiki(getWikiName(), true) + pageName;
   }
 
-  public String getUrlForPage(String pageName) {
+  public String getUrlForWikiPage(String pageName) {
     return getUrlForWiki() + pageName;
   }
 
-  public String getUrlForPage(String wikiName, String pageName) {
-    return getUrlForWiki(wikiName) + pageName;
-  }
-
-  public String getUrlForPage(Page page) {
-    if (page.getWikiPath() == null) {
-      return getUrlForWiki(page.getWikiName(), page.getWikiLanguage(), false);
-    }
-    return getUrlForPath(page.getWikiName(), page.getWikiLanguage(), page.getWikiPath());
-  }
-
-  public String getUrlForPage(Page page, String qs) {
-    return appendQueryStringToURL(page.getUrl(), qs);
-  }
-
-  public String getUrlForPath(String wikiName, String language, String wikiPath) {
-    return addPathToUrl(getUrlForWiki(wikiName, language, false, envType), wikiPath);
-  }
-
-  public String getUrlForPath(String wikiName, String wikiPath) {
-    return addPathToUrl(getUrlForWiki(wikiName), wikiPath);
-  }
-
   public String getUrlForPath(String wikiPath) {
-    return getUrlForPath(Configuration.getWikiName(), getLanguageForWiki(), wikiPath);
-  }
-
-  /** It actually adds string to url, so the path might consist a query
-  @param  url wiki's url.
-  @param  path path to be added to the url
-   **/
-  protected String addPathToUrl(String url, String path) {
-
-    String outputUrl = (!path.startsWith("/")) ? String.format("%s/%s", url, path): String.format("%s%s", url, path);
-
-    String qs = Configuration.getQS();
-    if (StringUtils.isNotBlank(qs)) {
-      outputUrl = appendQueryStringToURL(outputUrl, qs);
-    }
-    return outputUrl;
+    return addPathToUrl(getUrlForWiki(), wikiPath);
   }
 
   public String getUrlForWiki() {
-    return getUrlForWiki(Configuration.getWikiName(), getLanguageForWiki(), false);
-  }
-
-  public String getUrlForWiki(String wikiName, String language, boolean addWWW) {
-    return getUrlForWiki(wikiName, language, addWWW, envType);
+    return getUrlForWiki(this.language, false);
   }
 
-  public String getUrlForWiki(String wikiName) {
-    return getUrlForWiki(wikiName, false);
+  public String getUrlForWiki(String language, boolean addWWW) {
+    return getUrlForWiki(language, addWWW, envType);
   }
 
-  public String getUrlForWiki(String wikiName, boolean addWWW) {
+  public String getUrlForWiki(boolean addWWW) {
```

Comment:
Do we actually need the `addWWW` param? Please note that if will not work with https - `muppet.wikia.com` is covered by the certificate, but `www.muppet.wikia.com` is not.